internal/migrations/migrator.go

```diff
@@ -307,6 +295,65 @@ func (m *Migrator) NotifyHeight(ctx context.Context, block *common.BlockContext,
 	return nil
 }
 
+// generateGenesisConfig generates the genesis config for the new chain based on the snapshot and the current
+// chain's consensus params. It saves the genesis file to the migrations/snapshots directory.
+// This function is called only once at the start height of the migration.
+// It is run asynchronously as we don't have access to the cometbft's state during replay.
+// Therefore we need to wait for the consensus params fn to be set before we can generate the genesis file.
+func (m *Migrator) generateGenesisConfig(ctx context.Context, cfgParams *common.NetworkParameters, snapshotHash []byte, genesisValidators []*chain.GenesisValidator, logger log.Logger) error {
+	// block until the m.consensusParamsFn is set
+	<-m.consensusParamsFnChan
+
+	// sanity check
+	if m.consensusParamsFn == nil {
+		return fmt.Errorf("consensus params getter not set")
+	}
+
+	logger.Info("generating genesis config for the new chain", log.String("ChainID", m.activeMigration.ChainID))
+
+	height := m.activeMigration.StartHeight - 1
+	consensusParmas := m.consensusParamsFn(ctx, &height)
+	if consensusParmas == nil {
+		return fmt.Errorf("failed to get consensus params from cometbft")
+	}
+
+	genCfg := chain.DefaultGenesisConfig()
+	genCfg.ChainID = m.activeMigration.ChainID
+	// Initial height set to 0
+	genCfg.DataAppHash = snapshotHash
+	genCfg.Validators = genesisValidators
+	// Allocs are not needed, as the transfers are included in the snapshot
+	// forks can be dropped, as they maynot be relevant to the new chain
+
+	// Block Params
+	genCfg.ConsensusParams.BaseConsensusParams.Block.MaxBytes = cfgParams.MaxBlockSize
+	genCfg.ConsensusParams.BaseConsensusParams.Block.MaxGas = consensusParmas.Block.MaxGas
+	// Evidence Params
+	genCfg.ConsensusParams.BaseConsensusParams.Evidence = chain.EvidenceParams(consensusParmas.Evidence)
```

Comment:
I think this is alright, but we're essentially reversing `func ExtractConsensusParams(cp *chain.BaseConsensusParams) *cmtTypes.ConsensusParams {` (in internal/abci/cometbft/genesis.go`).

It seems like if we did this right, we wouldn't use `chain.DefaultGenesisConfig`, and we'd have a reverse of that function that takes the `cmtTypes.ConsensusParams` and the `common.NetworkParameters` to give our `chain.ConsensusParams.

I think combing things as above would be beneficial.